services/filemanager/azureBlobStoragemanager.go

```diff
@@ -134,15 +134,29 @@ func (manager *AzureBlobStorageManager) ListFilesWithPrefix(ctx context.Context,
 	defer cancel()
 
 	// List the blobs in the container
-	var marker string
-	response, err := containerURL.ListBlobsFlatSegment(ctx, azblob.Marker{Val: &marker}, segmentOptions)
-	if err != nil {
-		return
+	var response *azblob.ListBlobsFlatSegmentResponse
+	var startAfterTime time.Time
+	if manager.Config.StartAfter != "" {
+		startAfterTime, err = time.Parse(time.RFC3339, manager.Config.StartAfter)
+		if err != nil {
+			return
+		}
 	}
 
-	fileObjects = make([]*FileObject, len(response.Segment.BlobItems))
-	for idx := range response.Segment.BlobItems {
-		fileObjects[idx] = &FileObject{response.Segment.BlobItems[idx].Name, response.Segment.BlobItems[idx].Properties.LastModified}
+	for maxItems > 0 && manager.Config.Marker.NotDone() {
```

Comment:
Two comments here:

1. Why do we only do this for `maxItems > 0`? I believe we should always do this. 
> // List the blob(s) in our container; since a container may hold millions of blobs, this is done 1 segment at a time.


2. 
What if we follow this approach: 
```suggestion
	for marker := (Marker{}); marker.NotDone(); {
```

Based on the example:
```go
for marker := (Marker{}); marker.NotDone(); { // The parens around Marker{} are required to avoid compiler error.
	// Get a result segment starting with the blob indicated by the current Marker.
	listBlob, err := containerURL.ListBlobsFlatSegment(ctx, marker, ListBlobsSegmentOptions{})
	if err != nil {
		log.Fatal(err)
	}
	// IMPORTANT: ListBlobs returns the start of the next segment; you MUST use this to get
	// the next segment (after processing the current result segment).
	marker = listBlob.NextMarker

	// Process the blobs returned in this result segment (if the segment is empty, the loop body won't execute)
	for _, blobInfo := range listBlob.Segment.BlobItems {
		fmt.Print("Blob name: " + blobInfo.Name + "\n")
	}
}

```

Reply:
I understand that when `maxItems` is passed, we want to limit the number of returns items and used the filemanager as an iterator. 

However, given that the manager can be concurrently used by multiple go routines, it not safe to store the state of iteration in the manager :/